File: src/test/java/de/zalando/zally/rules/AvoidJavascriptKeywordsRuleTest.java
package de.zalando.zally.rules;

import io.swagger.models.Swagger;
import io.swagger.parser.SwaggerParser;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

import static org.assertj.core.api.Assertions.assertThat;

public class AvoidJavascriptKeywordsRuleTest {
    private Swagger validSwagger = getFixture("fixtures/avoidJavascriptValid.json");
    private Swagger invalidSwagger = getFixture("fixtures/avoidJavascriptInvalid.json");

    @Test
    public void positiveCase() {
        assertThat(new AvoidJavascriptKeywordsRule().validate(validSwagger)).isEmpty();
    }

    @Test
    public void negativeCase () {
        List<String> results = new AvoidJavascriptKeywordsRule()
                .validate(invalidSwagger)
                .stream()
                .map(v -> v.getPath().get())
                .collect(Collectors.toList());
        assertThat(results).hasSameElementsAs(Arrays.asList("Pet.return", "Pet.typeof"));
    }

    private Swagger getFixture(String fixture) {
        return new SwaggerParser().read(fixture);
    }
}

File: server/src/test/java/de/zalando/zally/rules/AvoidJavascriptKeywordsRuleTest.kt
package de.zalando.zally.rules

import de.zalando.zally.getFixture
import de.zalando.zally.testConfig
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class AvoidJavascriptKeywordsRuleTest {

    @Test
    fun positiveCase() {
        assertThat(AvoidJavascriptKeywordsRule(testConfig).validate(getFixture("avoidJavascriptValid.json"))).isNull()
    }

    @Test
    fun negativeCase() {
        val result = AvoidJavascriptKeywordsRule(testConfig).validate(getFixture("avoidJavascriptInvalid.json"))!!
        assertThat(result.paths).hasSameElementsAs(listOf("#/definitions/Pet"))
        assertThat(result.description).contains("return", "typeof")
    }
}
